File: solutions/0868. Binary Gap/0868_2.py
# Time Complexity: O(log⁡N).
# Space Complexity: O(1).
class Solution:
  def binaryGap(self, n: int) -> int:
    last = None
    ans = 0
    for i in range(32):
      if (n >> i) & 1:
        if last is not None:
          ans = max(ans, i - last)
        last = i
    return ans

File: solutions/0868. Binary Gap/0868_2.cpp
class Solution {
public:
  int binaryGap(int n) {
     if ((n & (n - 1)) == 0) {
      return 0;
     }
     n |= n - 1;
     int ans = 1;
     while ((n & (n + 1)) != 0) {
      n |= n >> 1;
      ans++;
     }    
     return ans;
  }
};